### PR TITLE
create addLabel controllers

### PR DESCRIPTION
**Pull request by ZhuoxiLi**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func main() {
 
 	gvkLabelMap := map[schema.GroupVersionKind]cache.Selector{
 		corev1.SchemeGroupVersion.WithKind("Secret"): {
-			FieldSelector: constants.SecretTypeTLS,
+			LabelSelector: constants.SecretWatchLabel,
```
Comment (ZhuoxiLi) [inline main.go:95]: Updated the `Cache Filter` filter by Label

File: controllers/cert-manager/v1_add_label_controller.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		return ctrl.Result{}, err
+	}
+
+	// Get secret corresponding to the certificate
+	secretInstance, err := r.getSecret(cert)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
```
Comment (horis233) [inline controllers/cert-manager/v1_add_label_controller.go:76]: Could we report a log when the error happens?

File: controllers/cert-manager/v1_add_label_controller.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	if oldLabelsMap == nil {
+		oldLabelsMap = make(map[string]string)
+	}
+	if _, ok := oldLabelsMap[res.SecretWatchLabel]; !ok {
+		oldLabelsMap[res.SecretWatchLabel] = ""
+		secretInstance.SetLabels(oldLabelsMap)
+	}
+
+	r.updateSecret(secretInstance)
+	return ctrl.Result{}, nil
```
Comment (bitscuit) [inline controllers/cert-manager/v1_add_label_controller.go:91]: Need to do something with the error, possibly return it and controller will then reconcile again

```suggestion
	return ctrl.Result{}, r.updateSecret(secretInstance)
```

File: controllers/cert-manager/v1alpha1_add_label_controller.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+		oldLabelsMap = make(map[string]string)
+	}
+
+	if _, ok := oldLabelsMap[res.SecretWatchLabel]; !ok {
+		oldLabelsMap[res.SecretWatchLabel] = ""
+		secretInstance.SetLabels(oldLabelsMap)
+	}
+
+	r.updateSecret(secretInstance)
+	return ctrl.Result{}, nil
```
Comment (bitscuit) [inline controllers/cert-manager/v1alpha1_add_label_controller.go:103]: Need to do something with the error
```suggestion
	return ctrl.Result{}, r.updateSecret(secretInstance)
```

**Comment by bitscuit** [comment]
other than my comments about handling the update secret error, PR is good and can be merged

**Comment by bitscuit** [comment]
I can see label is added to certificate secret
image

**Comment by horis233** [comment]
Let's also print the error out if `r.updateSecret(secretInstance)` return an error.
@zhuoxi

**Comment by ZhuoxiLi** [comment]
@horis233 this pr is ready for final review, thanks!

**Review by horis233 — APPROVED** [review]
/lgtm

**Pull request merged**